subsys/bluetooth/host/hci_core.c

```diff
@@ -802,6 +834,13 @@ static void le_conn_complete(struct net_buf *buf)
 		}
 	}
 
+	if (BT_FEAT_LE_DLE(bt_dev.le.features)) {
+		err = hci_le_set_data_len(conn);
+		if (!err) {
+			goto done;
```

Comment:
yes, if err  == 0, then wait for data_len_change event to arrive and proceed to setting phy in that event function.